### DISCUSSION TITLE
Editing or using a custom template for a Quarto html page

### DISCUSSION THREAD

**Discussion by ma-riviere** [quarto-dev/quarto-cli#2890] [Q&A] [answered]
Hi everyone,

How should one go about editing or using a custom template for a Quarto HTML page ?

**Fictive example:**
Let's say I have a personal website with a publication section. Each individual page for a publication has a header with the Authors, the DOI, the Date, etc. 
I would like to add another element to that header, e.g. "Funded by", which would display the name of the funding instrument and redirect to the funding instrument's webpage if you click on it.
Ideally, I'd want to add a `funding: NIH_grant_code` line to the .qmd of that article, and customize the template that parses that YAML to display the funding instrument (and possibly create automated links when you click on it, like what is done with the DOI).

Where can I find the template to edit to achieve this, and what's the best way to go about achieving this ? (i.e. maybe avoid editing the default templates inside the quarto-cli install, which will get overridden at each subsequent update).

Thank you !

**Comment by mcanouil** [comment, marked as answer]
This seems to be a custom listing (see <https://quarto.org/docs/websites/website-listings-custom.html>).

Also I started to work on some custom listing based on a question on Twitter by @allisonhorst (see <https://twitter.com/allison_horst/status/1580706829696348161?s=46&t=cl2CHnu2srfxxb_-_G_gMg>).

The WIP repository: https://github.com/mcanouil/quarto-grid-listing

PS: this seems to be more of a Q&A discussion than a general discussion, if that's the case could you change the category? (Only possible on the desktop version of the web page).

**Reply by ma-riviere** [reply]
I thought custom listings only applied to listing pages ? (e.g. the page where all the publications are listed, not the individual page of a publication)

**Reply by mcanouil** [reply]
I am not entirely sure what you really want to achieve.
If the listing does not answer your needs, I would suggest to build a template extension to use your own partial for the metadata (original partial <https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/templates/title-metadata.html>) and a Lua filter if needed.
<https://quarto.org/docs/journals/templates.html#html-partials>

**Reply by ma-riviere** [reply]
What I want is to be able to add things to the page's block, like this:

![image](https://user-images.githubusercontent.com/36962105/196405374-9903409c-b1e9-4fd0-a6ac-fb49f9c70305.png)

The title-metadata partial seems to be the one I need, thanks ! I found a few examples of HTML template extensions on your awesome-list, so it seems I have everything I need to get started.

Marking as solved, and moving it to Q&A like you suggested.
